### ISSUE TITLE
Open in editor on Markdown `.md` file not working

### ISSUE THREAD

**Issue by adriangalilea** [warpdotdev/Warp#4920]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

I open a Markdown file, press open in editor, and it simply opens a new tab with the same md file

### To reproduce

.

### Operating system

MacOS

### Operating system and version

14.4.1 (23E224)

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hi @adriangalilea sorry to hear about this. The editor option opens the default code editor in your `Settings > Features > Choose an editor to open file links`

https://github.com/warpdotdev/Warp/assets/16809145/54c294a1-d94f-4cd9-8eed-7c6e6e00b19b

Could you provide video or screenshots of the issue? Are you trying to open a markdown file from another md file?

**Comment by adriangalilea** [comment]
Ahhh, I see now, I thought it would use `$EDITOR`,

I think i'ts bad UX to read  my markdown files in the terminal but not be able to also edit them there, I would like to use my `$EDITOR` in this case Helix.

https://github.com/warpdotdev/Warp/assets/90320947/163abb5d-e4a1-4b44-ad0a-6f0f799a50a4

Here is the video just in case, but I'm guessing that it simply doesn't support terminal editors, very weird considering it's a terminal.

**Comment by dannyneira** [comment]
Yes, currently only GUI-based editors are supported and they need to be set as default in the Mac system settings. That being said, there is a request to open the files with CLI based editors we are tracking here: https://github.com/warpdotdev/Warp/issues/2320

As a possible workaround, I recommend you type in `$EDITOR <path/to/file>` to open the files in the files with your default terminal editor. 

As for the Markdown Viewer feature in Warp, it allows you to view markdown files directly in your terminal. All commands in your files will be rendered as interactive code blocks, and with a simple click (or CMD - Enter) the commands will be executed directly in the terminal. However, it currently does not support editing markdown files directly in the terminal.

I agree that being able to edit markdown files directly in the terminal using your preferred terminal-based editor would be a great feature (a code editor built into Warp is something we'd eventually want to support, no eta on this thou). I recommend subscribing and 👍 +1: the issue listed above to show interest in it.

**Comment by dannyneira** [comment]
Closing as this is a known limitation with existing feature requests.

**Comment by andershaig** [comment]
I have the same issue as above and believe this behavior changed over the past month or so. Previously I was able to open .md files in One Markdown by leaving it set as default.

**Comment by PaulRBerg** [comment]
Hey @dannyneira could you please re-open this issue? It's a bug and it's now occurring again

**Comment by dannyneira** [comment]
Understood, this may be a regression as I was able to reproduce this. Thanks for letting us know. Reopening till we can get a resolution.
